src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private T[] values;
+    private int size = 0;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_SIZE];
+    }
+
+    public void resizeArray() {
+        int newSize = (size + (size >> 1));
+        values = Arrays.copyOf(values, newSize);
+    }
+
+    public void checkIndex(int index) throws ArrayListIndexOutOfBoundsException {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("This index doesn't exist");
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size == values.length) {
+            resizeArray();
+        }
+        values[size] = value;
+        size++;
 
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("This index doesn't exist");
+        }
+
+        if (size == values.length) {
+            resizeArray();
+        }
+
+        System.arraycopy(values, index, values, index + 1, size - index);
```

Comment:
same, Use [System.arraycopy()](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#arraycopy-java.lang.Object-int-java.lang.Object-int-int-) to move your array elements.